linked_list/implementation.py

```diff
@@ -1,40 +1,127 @@
 from .interface import AbstractLinkedList
-
+from .node import Node
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        if elements:
+            for e in elements:
+                node = Node(e)
+                if not self.start:
+                    self.start = node
+                if self.end:
+                    self.end.next = node
+                self.end = node
 
     def __str__(self):
-        pass
+        str_list = []
```

Comment:
Oh, exactly! In fact, our team came up with this one liner:
`return '[{}]'.format(", ".join(str(p) for p in self.elements)) if self.elements else "[]"`
But we really should have used `iter()`!!!